Packs/SilentPush/Integrations/SilentPush/SilentPush.py

```diff
@@ -148,36 +417,319 @@ def test_module(client: Client) -> str:
+            'Age (days)': str(domain_info.get('age', 'N/A')),
+            'Risk Score': str(domain_info.get('sp_risk_score', 'N/A'))
+        }
+        markdown.append(tableToMarkdown('Domain Information', [basic_info]))
+
+        
+        risk_explain = str(domain_info.get('sp_risk_score_explain', 'N/A'))
+        if risk_explain != 'N/A':
+            markdown.append(f'### Risk Score Explanation\n{risk_explain}')
+
+     
+        whois_info = domain_info.get('whois_info', {})
+        if whois_info:
+            whois_info_list = [{'Key': k, 'Value': v} for k, v in whois_info.items()]
+            markdown.append(tableToMarkdown('WHOIS Information', whois_info_list))
+
+        markdown.append('\n---\n')  
+
+    return CommandResults(
+        outputs_prefix='SilentPush.Domain',
+        outputs_key_field='domain',
+        outputs=raw_response.get('domains', []),
+        readable_output='\n'.join(markdown),
+        raw_response=raw_response
+    )
+
+
+
+def get_domain_certificates_command(client: Client, args: dict) -> CommandResults:
+    """
+    Command handler for fetching domain certificate information.
     """
-    Command handler for fetching domain information.
+    domain = args.get('domain', 'silentpush.com')
+    demisto.debug(f'Processing certificates for domain: {domain}')
+
+    raw_response = client.get_domain_certificates(domain)
+    demisto.debug(f'Response from API: {raw_response}')
+
+    if 'error' in raw_response:
+        return CommandResults(
+            outputs_prefix='SilentPush.Certificates',
+            outputs_key_field='domain',
+            outputs=raw_response,
+            readable_output=f"Error: {raw_response['error']}",
+            raw_response=raw_response
+        )
+
+    readable_output = tableToMarkdown('Domain Certificates', raw_response)
+
+    return CommandResults(
+        outputs_prefix='SilentPush.Certificates',
+        outputs_key_field='domain',
+        outputs=raw_response,
+        readable_output=readable_output,
+        raw_response=raw_response
+    )
+    
+
+def search_domains_command(client: Client, args: dict) -> CommandResults:
+    query = args.get('query')
+    start_date = args.get('start_date')
+    end_date = args.get('end_date')
+    risk_score_min = arg_to_number(args.get('risk_score_min'))
+    risk_score_max = arg_to_number(args.get('risk_score_max'))
+    limit = arg_to_number(args.get('limit', 100))
     
-    This function processes the command for 'silentpush-list-domain-information', retrieves the
-    domain information using the client, and formats it for XSOAR output.
+    demisto.debug(f'Searching domains with query: {query}')
+
+    try:
+        raw_response = client.search_domains(
+            query=query,
+            start_date=start_date,
+            end_date=end_date,
+            risk_score_min=risk_score_min,
+            risk_score_max=risk_score_max,
+            limit=limit
```

Comment:
```suggestion
            limit=limit
```No return found, check the output